splunk_add_on_ucc_framework/app_manifest.py

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright 2021 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import warnings
+
+APP_MANIFEST_FILE_NAME = "app.manifest"
+APP_MANIFEST_WEBSITE = "https://dev.splunk.com/enterprise/reference/packagingtoolkit/pkgtoolkitappmanifest/"
+
+DEPRECATION_MESSAGE = f"""
+Comments are not allowed in app.manifest file.
+Please refer to {APP_MANIFEST_WEBSITE}
+"""
+
+
+class AppManifestFormatException(Exception):
+    pass
+
+
+class AppManifest:
+    def __init__(self):
+        self._manifest = None
+        self._comments = []
+
+    def get_addon_name(self) -> str:
+        return self._manifest["info"]["id"]["name"]
```

Comment:
Not sure if it is worth, but maybe when trying to access self._manifest we might consider to do:
try:
  return self._manifest["info"]["id"]["name"]
except KeyError as error:
  raise AppManifestFormatException

But again, I am not sure if we need to be that strict. I'll leave it to you :)

Reply:
That's a good suggestion, but I'd say that we can to introduce / reuse an existing checker for `app.manifest` file. We can work on that later.

It will still throw an error if that field is not present.